### PR TITLE
PF-744 consolidate config into write-config script

### PR DESCRIPTION
**Pull request by ddietterich**
This PR removes the render scripts and directory from the `integration` project and changes it to use the common `config/` directory in the top-level build.

I made some improvements to the scripting as well: using proper temp files and better handling of missing secrets in vault.

### CODE REVIEW

File: scripts/write-config.sh [hunk 1]
```diff
@@ -136,69 +144,107 @@ if [ -e "${outputdir}/target.txt" ]; then
...
-instancename=$(jq -r '.name' "${fil}")
-instanceproject=$(jq -r '.project' "${fil}")
-instanceregion=$(jq -r '.region' "${fil}")
+vaultgetb64 "secret/dsde/terra/kernel/${k8senv}/${namespace}/workspace/sqlproxy-sa" "${outputdir}/sqlproxy-sa.json"
+tmpfile=$(mktemp)
+vaultget "secret/dsde/terra/kernel/${k8senv}/${namespace}/workspace/postgres/instance" "${tmpfile}"
+instancename=$(jq -r '.name' "${tmpfile}")
+instanceproject=$(jq -r '.project' "${tmpfile}")
+instanceregion=$(jq -r '.region' "${tmpfile}")
 if [ "$instanceproject" == "null" ];
```
Comment (jaycarlton) [inline scripts/write-config.sh:240]: do you want single equals here?

File: .github/actions/integration-test/action.yml [hunk 2]
```diff
@@ -30,8 +19,6 @@ runs:
       run: |
         cd integration
         echo "Test Runner config"
```
Comment (jaycarlton) [inline .github/actions/integration-test/action.yml:21]: Why don't you need to call the new script from here?

File: scripts/write-config.sh [hunk 3]
```diff
@@ -136,69 +144,107 @@ if [ -e "${outputdir}/target.txt" ]; then
...
-vaultgetb64 "secret/dsde/terra/kernel/integration/tools/buffer/client-sa" "buffer-client-sa.json"
-vaultgetb64 "secret/dsde/terra/kernel/integration/common/testrunner/testrunner-sa" "testrunner-sa.json"
+# Janitor SA is only available in integration
+if [ "${k8senv}" = "integration" ]; then
+    vaultgetb64 "secret/dsde/terra/kernel/integration/tools/crl_janitor/client-sa" "${outputdir}/janitor-client-sa.json"
+else
+    echo "No janitor credentials for target ${target}"
+fi
+
+# Buffer SA naming is irregular
```
Comment (jaycarlton) [inline scripts/write-config.sh:198]: Can we clean this up in vault?

**Review by jaycarlton — APPROVED** [review]

**Review by ungwudik — APPROVED** [review]
LGTM. It looks like Unit and Connected tests did **not** run, though.
Error is `.github/workflows/test.yml#L116The workflow is not valid. .github/workflows/test.yml (Line: 116, Col: 9): 'target' is already defined`

https://github.com/DataBiosphere/terra-workspace-manager/actions/runs/899621811

Looks like it broke earlier. The first occurrence I see is in this PR that merged earlier https://github.com/DataBiosphere/terra-workspace-manager/pull/364/checks

Something for us to follow up on outside of this PR: being able to identify a failure to _kick off_ a GitHub check/workflow (in contrast to identifying a failure _during_ the workflow run).

**Comment by ddietterich** [comment]
I added the fix to `test.yml` to this PR. Assuming it runs and the tests pass, I'll merge this. I'm puzzled how that made it in.
Thanks for debugging @ungwudik!

**Comment by ungwudik** [comment]
> I added the fix to `test.yml` to this PR. Assuming it runs and the tests pass, I'll merge this. I'm puzzled how that made it in.
> Thanks for debugging @ungwudik!

Looks good 👍

**Pull request merged**